samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/SectionAwareServiceHelperImpl.java

```diff
@@ -123,7 +124,42 @@ public List getGroupReleaseEnrollments(String siteid, String userUid, String pub
 
 		return enrollments;
 	}
-	
+
+	public List getAllGroupsReleaseEnrollments(String siteid, String userUid, String publishedAssessmentId) {
+		List availEnrollments = getAvailableEnrollments(siteid, userUid);
+		List enrollments = new ArrayList();
+
+		HashSet<String> membersInReleaseGroups = new HashSet<>(0);
+		Collection<Group> siteGroups = null;
+		try {
+			Site site = SiteService.getSite(siteid);
+			siteGroups = site.getGroupsWithMember(userUid);
+
+			if (siteGroups.isEmpty() && isUserAbleToGradeAll(siteid, userUid)) {
+				return availEnrollments;
+			}
+
+			Set<String> groupIdsSet = siteGroups.stream()
+				.map(Group::getId)
+				.collect(Collectors.toSet());
+
+			membersInReleaseGroups = new HashSet<>(site.getMembersInGroups(groupIdsSet));
+		}
+		catch (IdUnusedException ex) {
+			// no site found, just log a warning
+			log.warn("Unable to find a site with id ("+siteid+") in order to get the enrollments, will return 0 enrollments");
+		}
+
+		for (Iterator eIter = availEnrollments.iterator(); eIter.hasNext(); ) {
```

Comment:
really would avoid Iterator whenever possible. so much cleaner with nice `for`:

```
List<EnrollmentRecord> enrollments = new ArrayList<>();
for (EnrollmentRecord enr : availEnrollments) {
```

Reply:
@ottenhoff Totally agree. I copied from https://github.com/sakaiproject/sakai/pull/12681/files#diff-b1d3372a6dd0f8da1fede02cdfc1e1251b18ed62f2c223fd585a94c2137befd4R118

Should I change both?

Reply:
@ottenhoff Please, I need an answer on this. If I don't get it, it will be merged as is. Thanks!

Reply:
Fixed. Thanks!